x/leveragelp/keeper/hooks_amm.go

```diff
@@ -8,27 +8,13 @@ import (
 )
 
 func (k Keeper) CheckAmmPoolUsdcBalance(ctx sdk.Context, ammPool ammtypes.Pool) error {
-	leveragePool, found := k.GetPool(ctx, ammPool.PoolId)
-	if !found {
-		// It is possible that this pool haven't been enabled
-		return nil
-	}
-
-	// This is kind of health check so we should only use real amm pool balance
-	tvl, err := ammPool.TVL(ctx, k.oracleKeeper, nil)
-	if err != nil {
-		return err
-	}
-	leverageLpTvl := tvl.
-		Mul(leveragePool.LeveragedLpAmount.ToLegacyDec()).
-		Quo(ammPool.TotalShares.Amount.ToLegacyDec())
-
-	depositDenom := k.stableKeeper.GetDepositDenom(ctx)
-	price := k.oracleKeeper.GetAssetPriceFromDenom(ctx, depositDenom)
+	stablestakePool := k.stableKeeper.GetPool(ctx, ammPool.PoolId)
 
 	for _, asset := range ammPool.PoolAssets {
-		if asset.Token.Denom == depositDenom && price.MulInt(asset.Token.Amount).LT(leverageLpTvl) {
-			return types.ErrInsufficientUsdcAfterOp
+		for _, liabilties := range stablestakePool.TotalLiabilities {
+			if asset.Token.Denom == liabilties.Denom && asset.Token.Amount.LT(liabilties.Amount) {
+				return types.ErrInsufficientUsdcAfterOp
+			}
 		}
```

Comment:
This is wrong, I think we need to subtract current liabilities as that is already exited from amm pool and then check

Reply:
Yeah will do that in exit pool PR